core/src/main/java/it/pagopa/selfcare/mscore/core/SchedulerServiceImpl.java

```diff
@@ -90,11 +98,54 @@ private void sendDataLakeNotifications(List<Token> tokens) {
         });
     }
 
+    private void sendDataLakeUserNotifications(List<OnboardedUser> users, String productId){
+        users.forEach(onboardedUser -> {
+            userEventService.sendOnboardedUserNotification(onboardedUser, productId);
+        });
+    }
+
+    @Async
+    public void regenerateUserNotifications(Optional<String> userId){
+        if (productsFilter.isPresent()){
```

Comment:
what's happen if not productsFilter is defined?

Reply:
the product's filter is a required parameter in the api call, it cannot be null